### PR TITLE
Converted step-context and use-steps hook to typescript

### PR DESCRIPTION
**Pull request by mhevyk**
Converted `step-context` and `useSteps` hook to typescript

Previous implementation was hard to convert to typescript because of following function:

```jsx
const handleStepData = useCallback(
    (stepLabel, data, errors) => {
      switch (stepLabel) {
        case generalLabel:
          setGeneralData({ data, errors })
          break
        case subjectLabel:
          setSubject(data)
          break
        case languageLabel:
          setLanguage(data)
          break
        case photoLabel:
          setPhoto(data)
          break
        default:
          return
      }
    },
    [generalLabel, subjectLabel, languageLabel, photoLabel]
  )
```

In code snippet above `stepLabel` is just string and we can't use typescript's discriminated union feature here to correctly infer the type for each case.

So, I decided to use `useReducer` hook here instead of couple `useState`s to manage state more efficiently. We achived type safety depending on operation we want to handle. I also created pleasant interface for stepData setters, that looks like this:

**BEFORE:** 
```jsx
const onChangeLanguage = (_, value) => {
    handleStepData(stepLabel, value)
}
```

**AFTER:**
```jsx
const onChangeLanguage = (_, value) => {
    handleLanguage(value)
}
```

I removed step labels because they have only visual role inside tutor stepper and student stepper, but the state stored inside both steppers is the same

### CODE REVIEW

File: src/services/user-service.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const userService = {
   },
   updateUser: (
     userId: string,
-    params: UpdateUserParams
+    params?: UpdateUserParams
```
[the remark this answers is not in the document]
Comment (mhevyk) [reply src/services/user-service.tsx:28]: Update user with empty body returns empty response if I am not mistaken, so this supposal can be correct here

image

I would be pleased to hear your opinion on this

[the remark this answers is not in the document]
Comment (iva-stasia) [reply src/services/user-service.tsx:28]: What I meant is that it doesn't seem logical. I mean, what's the purpose of calling the `updateUser` function if we don't want to update any data? From that point, it seems better to leave `params` required. Why do we need to make it optional?

[the remark this answers is not in the document]
Comment (mhevyk) [reply src/services/user-service.tsx]: I agree with you that it would be better to keep `params` required in service. Fixed

**Review by iva-stasia — APPROVED** [review]

**Review by GeorgeShvab — APPROVED** [review]

**Review by D1nviz — APPROVED** [review]

File: src/types/user/user-interfaces/user.interfaces.ts [hunk 2]
```diff
@@ -42,6 +42,12 @@ export interface UserResponse {
   FAQ: DataByRole<Faq[]>
 }
 
+export interface UserGeneralInfo
+  extends Pick<UserResponse, 'firstName' | 'lastName' | 'professionalSummary'> {
+  country: UserResponse['address']['country'] | null
```
[the remark this answers is not in the document]
Comment (asya982) [reply src/types/user/user-interfaces/user.interfaces.ts:47]: I meant this type `export type Address = {
  country: string
  city: string
}` which is declared on 3 line in this file

**Review by maarekua — APPROVED** [review]

**Review by byavd — APPROVED** [review]

**Review by asya982 — APPROVED** [review]

**Pull request merged**